nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/serialization/jaxb/JAXBSerializer.java

```diff
@@ -88,9 +98,15 @@ public T deserialize(final InputStream input) throws SerializationException {
         try {
             // Consume the header bytes.
             readDataModelVersion(input);
+
+            XMLStreamReader streamReader = XMLInputFactory.newInstance().createXMLStreamReader(input);
```

Comment:
Is it necessary to create an `XMLStreamReader` as opposed to wrapping the input in a `StreamSource`?